lessons/04-structured-output/README.md

```diff
@@ -0,0 +1,590 @@
+
+**Leonardo:** "One more game," he says, his eyes gleaming with determination.  George (Time beetle), we need to win this one, give me the right moves.
+
+**Time beetle:** Very well, calculating... Here you go...
+
+The final game is a nail-biter. The Aztec soldiers and courtiers gather around, their eyes fixed on the board. Every move is met with gasps and murmurs of anticipation. Leonardo's hands move with precision and confidence, and as the game progresses, it becomes clear that he has the upper hand. 
+
+With a final, decisive move, Leonardo wins the game. The room erupts in cheers and applause, the Aztecs clearly impressed by his skill and composure. Montezuma, though disappointed, nods in respect. 
+
+**Montezuma:** "You have won fairly," Montezuma says, his voice carrying a note of admiration. "You are free to go, and you may keep your device. But know that you are always welcome here, should you wish to return." 
+
+**Time beetle:** I won't say anything if you don't Leonardo ;)
+ 
+> Patolli is one of the oldest known games in the Americas, played by various pre-Columbian Mesoamerican cultures, including the Aztecs. 
+> 
+> It is a game of strategy and luck, often associated with gambling. 
+>
+> How Patolli is Played: 
+>
+> Board and Pieces: The game is played on an X-shaped board with a center block and four arms. Players use five black beans as dice, each marked on one side 
+> 
+> Setup: Each player selects six markers (beads) of their color and places them off the board. Players also have point beads for scoring 
+> 
+> Starting the Game: Players roll the beans to move their markers around the board. The goal is to move all markers around the board and back to the starting point 
+> 
+> Movement: Players roll the beans to determine how many spaces to move. A roll of five allows a marker to move ten spaces 
+> 
+> Capturing and Scoring: Landing on specific spaces can result in gaining extra turns, losing point beads, or capturing opponent's markers 
+> 
+> Aztec Involvement: 
+> 
+> The Aztecs, including nobles and commoners, widely played Patolli. It was a popular pastime at the court of Montezuma, who enjoyed watching his nobles play 
+> 
+> The game was not just for entertainment but also involved high-stakes gambling, with players betting items like blankets, precious stones, and even their homes or freedom 
+
+![Patolli board](./assets/patolli.png)
+
+*Patolli - Wikipedia*
+
+## Structured Output
+
+**Time beetle:** Hey folks, want to know how I beat Montezuma?
+
+**You:** Sshh, not so loud, we're still in the Aztec Empire.
+
+**Time beetle:** Oh right, sorry, yea so structured output is how. You asked for the right moves and I gave them to you in a structured way.
+
+Structured output refers to different data formats that are used to present information in a structured manner. Formats like JSON, XML, and CSV are commonly used and makes it easy to consume data by services.
```

Comment:
Might simplify this a little to something like:

Structured output uses specific data formats to organize information clearly. Common formats like JSON, XML, and CSV make it easier for services to process and use the data.